src/chains/ethereum/ethereum/tests/api/eth/getBlockByNumber.test.ts

```diff
@@ -97,6 +99,288 @@ describe("api", () => {
+          );
+        };
+
+        describe("basic checks", () => {
+          beforeEach(async () => {
+            provider = await getProvider({});
+            [from, to] = await provider.send("eth_accounts");
+          });
+          afterEach(async () => {
+            provider && (await provider.disconnect());
+          });
+
+          it(`has a block number of "latest" block + 1`, async () => {
+            const pendingBlock = await provider.send("eth_getBlockByNumber", [
+              "pending"
+            ]);
+            const latestBlock = await provider.send("eth_getBlockByNumber", [
+              "latest"
+            ]);
+            assert.strictEqual(
+              Quantity.toNumber(pendingBlock.number),
+              Quantity.toNumber(latestBlock.number) + 1,
+              `Pending block doesn't have expected number.`
+            );
+          });
+
+          it(`has the correct \`baseFeePerGas\``, async () => {
+            const pendingBlock = await provider.send("eth_getBlockByNumber", [
+              "pending"
+            ]);
+            const latestBlock = await provider.send("eth_getBlockByNumber", [
+              "latest"
+            ]);
+            // baseFeePerGas is .875 (aka 7/8) of the previous block if empty
+            const expected =
+              (Quantity.toBigInt(latestBlock.baseFeePerGas) * 7n) / 8n;
+            assert.strictEqual(
+              Quantity.toBigInt(pendingBlock.baseFeePerGas),
+              expected,
+              `Pending block doesn't have expected \`baseFeePerGas\`.`
+            );
+          });
+
+          it(`has no transactions if the pool is empty`, async () => {
+            const pendingBlock = await provider.send("eth_getBlockByNumber", [
+              "pending"
+            ]);
+            assert.strictEqual(
+              pendingBlock.transactions.length,
+              0,
+              `Pending block isn't empty when the transaction pool was empty.`
+            );
+          });
+        });
+
+        describe("`miner.blockTime=0` mode", () => {
+          beforeEach(async () => {
+            provider = await getProvider({
+              miner: {
+                // strict mode doesn't affect the contents of the pending block,
+                // but it will make it easier to fetch a pending block before
+                // transactions start mining
+                instamine: "strict",
+                blockGasLimit
+              }
+            });
+            [from, to] = await provider.send("eth_accounts");
+          });
+
+          afterEach(async () => {
+            provider && (await provider.disconnect());
+          });
+
+          it(`has one transaction regardless of the number of executables in the pool and is equal to next mined block`, async () => {
+            // send 1 more transaction than will fit in the block
+            for (let i = 0; i < emptyTransactionsPerBlock + 1; i++) {
```

Comment:
Yes, I think you're right. That's why I limit it to only a few transactions :)

I suppose for this test I could just send two transactions, because I'm just proving that it will only add one transaction to the block. Hmm

Reply:
If we use `memdown` for our db in this test I think it would be (more) deterministic -- at least in regards to the timing of when internal promises are resolved, and would no longer be at the whims of disk IO. If this test could possibly fail due to faster/slower I/O then we should remedy that.

Reply:
I've implemented `memdown` as suggested